File: java/CalculadoraExemplo/src/test/java/com/mycompany/app/CalculadoraTest.java
package com.mycompany.app;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;


public class CalculadoraTest {
    
    private Calculadora calculadora;
    
    public CalculadoraTest() {
        
        this.calculadora = new Calculadora();
    }
    
    @BeforeEach
    public void setUp() {
        this.calculadora.setResultado(0.0);
    }

    @Test
    public void testSomar() {
        double a = 2.0;
        double b = 2.0;
        this.calculadora.somar(a, b);
        assertEquals(4.0, this.calculadora.getResultado());
    }

    @Test
    public void testSubtrair() {
       double a = 3.0;
        double b = 2.0;
        this.calculadora.subtrair(a, b);
        assertEquals(1.0, this.calculadora.getResultado());
    }

    @Test
    public void testMultiplicar() {
        double a = 2.0;
        double b = 2.0;
        this.calculadora.multiplicar(a, b);
        assertEquals(4.0, this.calculadora.getResultado());
    }

    @Test
    public void testDividirValoresComuns() {
        double a = 4.0;
        double b = 4.0;
        this.calculadora.dividir(a, b);
        assertEquals(1.0, this.calculadora.getResultado());
    }
    
    @Test
    public void testDivisaoPorZeroExcecao() {
        double a = 4.0;
        double b = 0.0;
        Exception exception = assertThrows(ArithmeticException.class, () ->
            this.calculadora.dividir(a, b));
        assertEquals("Impossivel realizar operacao de divisao por zero", exception.getMessage());
    }
}

File: php/calculadoraphp/tests/CalculadoraTest.php
<?php 

require_once(__DIR__ . '/../vendor/autoload.php');

use PHPUnit\Framework\TestCase;
use App\Calculadora;

final class CalculadoraTest extends TestCase
{
    private $calculadora;

    /**
     * @before
     */
    protected function setUp() : void
    {
        $this->calculadora = new Calculadora();
    }

    /**
     * @test
     */
    public function testSomarDoisElementos() : void
    {
        $a = 2.0;
        $b = 4.0;
        $this->calculadora->somar($a, $b);
        $this->assertEquals(4.0, $this->calculadora->getResultado());
    }

    /**
     * @test
     */
    public function testSubtrairDoisElementos() : void
    {
        $a = 4.0;
        $b = 2.0;
        $this->calculadora->subtrair($a, $b);
        $this->assertEquals(2.0, $this->calculadora->getResultado());
    }

    /**
     * @test
     */
    public function testMultiplicarDoisElementos() : void
    {
        $a = 4.0;
        $b = 2.0;
        $this->calculadora->multiplicar($a, $b);
        $this->assertEquals(8.0, $this->calculadora->getResultado());
    }

    /**
     * @test
     */
    public function testDividirDoisElementos() : void
    {
        $a = 4.0;
        $b = 4.0;
        $this->calculadora->dividir($a, $b);
        $this->assertEquals(1.0, $this->calculadora->getResultado());
    }

    /**
     * @test
     */
    public function testDivisaoPorZero() : void
    {
        $a = 4.0;
        $b = 0.0;
        $this->expectException(ArithmeticError::class);
        $this->expectErrorMessage("Impossivel operacao de divisao por zero");
        $this->calculadora->dividir($a, $b);
    }
}